### PR TITLE
Improve hashing xp

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by jrapin** [comment]
is this still used at all? I would be in favor of a dedicated ExperimentFunction for hashing

File: nevergrad/functions/functionlib.py
Comment (jrapin) [inline nevergrad/functions/functionlib.py:54]:
```suggestion
            for y in data:
                np.random.seed(int(hashlib.md5(str(y).encode()).hexdigest(), 16) % 500000)  # type: ignore
```

**Review by jrapin — APPROVED** [review]
I'm still in favor of externalizing this part of the code. It could be an ExperimentFunction taking another experiment function as input I guess.
Also, this is not efficient at all, but I guess nevermind.

See the change to do before merging.

**Comment by teytaud** [comment]
Regarding whether we use it: we don't use it because this is buggy. Without the bug, this is an infinite discrete domain: definitely worth it.

I see your point regarding a possible distinct ExperimentFunction: I'll see that, I do not merge for now.

**Comment by teytaud** [comment]
Ok, let's reduce the list of PR, I merge.

**Pull request merged**
